src/Npgsql/TypeHandlers/CompositeHandlers/ByReference.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Diagnostics;
+
+namespace Npgsql.TypeHandlers.CompositeHandlers
+{
+    sealed class ByReference<T>
```

Comment:
nit: constraint T to struct (mostly for documentation value), this would allow you to remove the `Debug.Assert()` down below.

Reply:
Not possible without return of `Unsafe` which was removed by @khellang. This type is used in `MappedCompositeHandler<T>` which doesn't have a constraint on `T`.

Reply:
I don't think I've ever touched this type? 🤔

Reply:
Makes sense - at least the part where it's used by `MappedCompositeHandler<T>` which itself isn't constrained. It's not important.

Reply:
@khellang You have removed `Unsafe` via #2310 and #2313.

Reply:
Right, but not in this type. There are still valid use cases for `Unsafe` 😉